addons/point_of_sale/static/src/app/screens/product_screen/order_summary/order_summary.js

```diff
@@ -61,6 +61,21 @@ export class OrderSummary extends Component {
     async updateSelectedOrderline({ buffer, key }) {
         const order = this.pos.get_order();
         const selectedLine = order.get_selected_orderline();
+        // Handling negation of value on first input
+        if (buffer === "firstInput" && key == "-") {
+            if (this.pos.numpadMode === "quantity") {
+                if (selectedLine.refunded_orderline_id) {
+                    // Negative amount not allowed for refunds
+                    this.numberBuffer.state.buffer = buffer = "-0";
```

Comment:
It allows for adjusting the quantity during a refund. For instance, if a product was originally ordered with a quantity of 3 and we have pressed refund with 1 qty now we have productscreen with -1 qty, you can’t directly input 3 to change it, as it will be considered as +3, resulting in a error popup stating that refund lines cannot have a positive quantity. 

To handle this, you would use the +/- button to adjust the quantity incrementally. Starting from -1, you can decrease it to -0 on pressing +/-, and then input 3 to reflect -03 = -3  the correct refund amount. This functionality ensures accurate quantity adjustments for refunds.